pipeline/test/test_pipeline.py

```diff
@@ -240,6 +282,21 @@ def test_single_task_write_resource_group(self):
         p.write_output(t.output.foo, f'{gcs_output_dir}/test_single_task_write_resource_group_file.txt')
         p.run()
 
+    def test_multiple_dependent_tasks(self):
+        output_file = f'{gcs_output_dir}/test_multiple_dependent_tasks.txt'
+        p = self.pipeline()
+        t = p.new_task()
+        t.command(f'echo "0" >> {t.ofile}')
```

Comment:
Can you help me understand where t.ofile is declared?

Reply:
Is it automatically added as a property by `command`?

Reply:
`__getattr__` for `t.ofile` and `__getitem__` for `t['ofile']`.

Reply:
It feels like this should be explicitly assigned, or not mentioned at all:

```
t.command(f'echo "0" >>')
```

or

```
command(f'echo "0" >> {t}')
```

with some `__str__`


I won't block the PR due to this, since that would touch outside functionality, but I think if we're specifying attributes, implicitly adding them as part of an attempt to read them is semantically, imo, incorrect, and will lead to difficult to debug behavior (unless the only way to add attributes is implicit, which it isn't, since we already have _new_resource_group).

I would recommend we have `declare_resource_group` as the only method to add attributes.

Reply:
Can you write a dev post with your concerns? We iterated quite a bit on the interface. Also it would be good to move this discussion outside of this PR.

Reply:
Sure

Reply:
https://dev.hail.is/t/pipeline-implicit-attribute-creation/137

(I couldn't cc 3 people)